tsconfig.build.json

```diff
@@ -0,0 +1,44 @@
+{
+  "compilerOptions": {
+    "assumeChangesOnlyAffectDirectDependencies": true,
+    "allowSyntheticDefaultImports": true,
+    "alwaysStrict": true,
+    "declaration": true,
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "composite": true,
+    "lib": [
+      "esnext"
+    ],
+    "module": "CommonJS",
+    "moduleResolution": "node",
+    "noImplicitThis": true,
+    "outDir": "dist",
+    "skipLibCheck": true,
+    "strict": true,
+    "strictNullChecks": true,
+    "rootDirs": [
+      "src/index.ts",
+      "src/types/BaalFactory.ts",
+      "src/types/BaalSummonerFactory.ts",
+      "src/types/LootFactory.ts",
+      "src/types/SharesFactory.ts",
+      "src/types/PosterFactory.ts",
+      "src/types/TributeMinionFactory.ts",
+      "abi/**/*.json"
+    ],
+    "target": "es6",
+  },
+  "include": [
+    "src/index.ts",
+    "src/types/BaalFactory.ts",
+    "src/types/BaalSummonerFactory.ts",
+    "src/types/LootFactory.ts",
+    "src/types/SharesFactory.ts",
+    "src/types/PosterFactory.ts",
+    "src/types/TributeMinionFactory.ts",
+    "abi/**/*.json"
```

Comment:
these config need to be maintained if you need want to add more types to the final dist